tools/src/main/java/org/apache/kafka/tools/reassign/ReassignPartitionsCommand.java

```diff
@@ -654,6 +654,13 @@ static Map<TopicPartition, List<Integer>> getReplicaAssignmentForPartitions(Admi
                     res.put(tp, info.replicas().stream().map(Node::id).collect(Collectors.toList()));
             })
         );
+
+        if (!res.keySet().equals(partitions)) {
+            Set<TopicPartition> missingPartitions = new HashSet<>(partitions);
+            missingPartitions.removeAll(res.keySet());
+            throw new UnknownTopicOrPartitionException("Unable to find partition: " +
+                missingPartitions.stream().map(TopicPartition::toString).collect(Collectors.joining(", ")));
```

Comment:
In ReassignPartitionsCommand#describeTopics, we wrap UnknownTopicOrPartitionException with ExecutionException. Maybe we should make it consistent by this pattern?

Reply:
Though I don't know why do we do this, but follow the pattern is a good choice. Thanks for the comment :smiley: